### ISSUE TITLE
Two unexpected behaviors of cupy.random.randint()

### ISSUE THREAD

**Issue by mogami290** [cupy/cupy#1529]
1, cupy.random.randint() makes an unexpected behavior to size=0.
I think it should behave the same way as numpy.
So too cupy.random.random_integer().
```
>>> import cupy
>>> import numpy
>>> numpy.random.randint(2432, size=0)
array([], dtype=int64)
>>> cupy.random.randint(2432, size=0)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/mogami/.pyenv/versions/anaconda3-4.3.1/lib/python3.6/site-packages/cupy/random/sample.py", line 112, in randint
    return rs.randint(low, high, size, dtype)
  File "/home/mogami/.pyenv/versions/anaconda3-4.3.1/lib/python3.6/site-packages/cupy/random/generator.py", line 471, in randint
    x = self.interval(diff, size).astype(dtype, copy=False)
  File "/home/mogami/.pyenv/versions/anaconda3-4.3.1/lib/python3.6/site-packages/cupy/random/generator.py", line 236, in interval
    return ret.reshape(size)
AttributeError: 'NoneType' object has no attribute 'reshape'
>>> cupy.__version__
'4.3.0'
```

2,  cupy.random.randint() makes another unexpected behavior to size=cupy.array(10).
I think it should behave the same way as numpy.
So too cupy.random.random_integer().
```
>>> np.random.randint(0xFFFFFFFF, size=np.array(10))
array([1769267337,  636691460,  721006452, 1931528634, 1669428804,
        686866511, 2434245456,  820877111, 3036591490,  486170555])
>>> 
>>> cupy.random.randint(0xFFFFFFFE, size=cupy.array(10))
array(3467725367)
>>> 
>>> cupy.random.randint(0xFFFFFFFE, size=10)
array([ 163488656, 4244709151, 2724177304, 1142413559, 2528432936,
       2369452189,  678685836, 2988871988,  783904484, 2443284494])
>>> 
>>> cupy.random.randint(0xFFFFFFFE, size=int(cupy.array(10)))
array([2203975300, 2743751689, 1639728575, 2964661434, 3078897270,
       2878417667, 2773421587,  450390496, 1177698536, 2323130837])
>>> 
>>> cupy.random.randint(0xFFFFFFFE, size=np.array(10))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/mogami/.pyenv/versions/anaconda3-4.3.1/lib/python3.6/site-packages/cupy/random/sample.py", line 112, in randint
    return rs.randint(low, high, size, dtype)
  File "/home/mogami/.pyenv/versions/anaconda3-4.3.1/lib/python3.6/site-packages/cupy/random/generator.py", line 471, in randint
    x = self.interval(diff, size).astype(dtype, copy=False)
  File "/home/mogami/.pyenv/versions/anaconda3-4.3.1/lib/python3.6/site-packages/cupy/random/generator.py", line 208, in interval
    n = functools.reduce(operator.mul, size, 1)
TypeError: reduce() arg 2 must support iteration
>>> 
>>> cupy.__version__
'4.3.0'
```

**Comment by JennieGerhardt** [comment]
do you have solved the first problem?

**Comment by toslunar** [comment]
The first problem is fixed by #2662.  Thanks.

**Comment by asi1024** [comment]
Moving discussion to #4831.
